app/models/message.rb

```diff
@@ -1,59 +1,89 @@
+# frozen_string_literal: true
+
 class Message < ApplicationRecord
-  acts_as_spammable fields: [ :subject, :body ],
+  acts_as_spammable fields: [:subject, :body],
     user: :from_user
 
-  blockable_by lambda { |message| message.to_user_id },
-    blockable_user_id: lambda { |message| message.from_user_id }
-  
-  requires_privilege :speech, if: Proc.new {|m|
+  blockable_by ->( message ) { message.to_user_id },
+    blockable_user_id: ->( message ) { message.from_user_id }
+
+  requires_privilege :speech, if: proc {| m |
     from_user_id == user_id && ( m.thread_id.blank? || m.thread_id == m.id )
   }
 
   belongs_to :user
-  belongs_to :from_user, :class_name => "User"
-  belongs_to :to_user, :class_name => "User"
+  belongs_to :from_user, class_name: "User"
+  belongs_to :to_user, class_name: "User"
 
   validates_presence_of :user
   validates :from_user_id, presence: true, numericality: { greater_than: 0 }
   validates :to_user_id, presence: true, numericality: { greater_than: 0 }
   validate :validate_to_not_from
-  validates :body, :presence => true
+  validates :body, presence: true
   before_create :set_read_at, :set_subject_for_reply
   after_save :set_thread_id
   after_create :deliver_email
-  
-  scope :inbox, -> { where("user_id = to_user_id") } #.select("DISTINCT ON (thread_id) messages.*")
-  scope :sent, -> { where("user_id = from_user_id") } #.select("DISTINCT ON (thread_id) messages.*")
-  scope :unread, -> { where("read_at IS NULL") }
+
+  scope :inbox, -> { where( "user_id = to_user_id" ) } # .select("DISTINCT ON (thread_id) messages.*")
+  scope :sent, -> { where( "user_id = from_user_id" ) } # .select("DISTINCT ON (thread_id) messages.*")
+  scope :unread, -> { where( "read_at IS NULL" ) }
 
   INBOX = "inbox"
   SENT = "sent"
-  BOXES = [INBOX, SENT]
+  BOXES = [INBOX, SENT].freeze
+
+  attr_accessor :html, :skip_email
+
+  # When we added sent_at, we could not populate it because it was impossible
+  # to distinguish between messages with no to_user_copy because the sender
+  # was suspended or because the receiver manually deleted their copy, so we
+  # need this crude date filter to prevent us from redelivering ancient
+  # messages that the recipient deleted.
+  RESEND_UNSENT_RELEASE_DATE = Date.parse( "2025-01-28" )
 
-  attr_accessor :html
-  attr_accessor :skip_email
+  def self.resend_unsent_for_user( user )
+    user = User.find_by_id( user ) unless user.is_a?( User )
+    return unless user
+
+    scope = user.messages.sent.
```

Comment:
The fact there is a scope named `sent`, and a new method named `sent?` with different meanings is a little confusing for readability. I was about to comment - should this say `.unsent.` - but this code seems right since its using the sent scope. Just a little strange that we're using the `sent` scope, and in the block below we skip the message if its `sent?`. Maybe worth a comment in cases where we use the two in close proximity?

Reply:
Yeah, not ideal. I started with "delivered" and then realized we were using that word to mean an email was delivered, so I went with "sent" to avoid confusion there, then realized we had that scope. How would you feel about renaming the scope to "outbox" instead? It's a scope that retrieves the user's copy of the messages they *tried* to send, not necessarily messages that were actually sent, so "outbox" might capture the meaning a little better, even if it breaks the email analogy a bit.

Reply:
I'm OK with renaming the scope to `outbox`. With any scope I usually need to go back to the definition to see what it _really_ means. I think `outbox` would be much better for at a glance readability, especially now that messages may not have been `sent` even though there is a copy from the sender.